### PR TITLE
Reconsent Tweaks & Results Report Content

### PR DESCRIPTION
**Pull request by cassidysymons**
- Adjustments to reconsent paths based on final testing
- Updates to results report content that were missed earlier

### CODE REVIEW

File: microsetta_interface/templates/new_results_page.jinja2 [hunk 1]
```diff
@@ -1363,11 +1366,11 @@
                 <img class="diversity-banner" src='/static/img/{{ _(EN_US_KEY) }}/hero-home-4-xl.jpeg'>
                 <br/><br />
                 <p>
-                    {{ _('Your microbiome is a rainforest of diverse microbes dominated by microbes - some 40 trillion bacteria, weighing around 1 lb, live all over and in your body. The diversity of your microbiome, how similar it is to other peoples\', and the types of bacteria living inside you have all been linked to health, disease and lifestyle traits in numerous studies.') }}
+                    {{ _('There are 40 trillion microbial cells in our bodies. This collection of microbial cells is known as the human microbiome. With roughly 3lbs (1.4kgs) of them in our gut alone, we\'re on a mission to make critical discoveries about their role in our lives. Thank you for contributing to our research project and helping advance the science.') }}
```
Comment (wasade) [inline microsetta_interface/templates/new_results_page.jinja2:1369]: The weight is not supported, see https://www.nature.com/articles/s41564-023-01426-7, which suggests ~1lb

File: microsetta_interface/templates/sample.jinja2 [hunk 2]
```diff
@@ -203,17 +216,17 @@
...
-                            <input id="sample_time" name="sample_time" class="form-control" type="text" value="{{ sample.time | e}}" autocomplete="off" required\>
+                            <input id="sample_time" name="sample_time" class="form-control" type="text" value="{{ sample.time | e}}" autocomplete="off" {% if (need_reconsent_data or need_reconsent_biospecimen) and not admin_mode %}disabled{% endif %} required\>
                         </div>
                     </div>
                     <div class="row mt-2">
                         <div class="col-12">
                             <label for="sample_site" name="sample_site_label" class="sample-label">{{ _('Sample Type') }}*</label>
-                            <select id="sample_site" name="sample_site" class="form-control" required {% if sample.sample_edit_locked and not admin_mode %}disabled{% endif %}>
+                            <select id="sample_site" name="sample_site" class="form-control" required {% if (sample.sample_edit_locked and not admin_mode) or ((need_reconsent_data or need_reconsent_biospecimen) and not admin_mode) %}disabled{% endif %}>
```
Comment (wasade) [inline microsetta_interface/templates/sample.jinja2:229]: `admin_mode` test is performed twice

File: microsetta_interface/templates/sample.jinja2 [hunk 3]
```diff
@@ -228,7 +241,7 @@
                     <div class="row mt-2">
                         <div class="col-12">
                             <label for="sample_notes" id="sample_notes_label" class="sample-label">{{ _('Notes') }}</label>
-                            <textarea class="form-control" id="sample_notes" name="sample_notes" placeholder="{{ _('(Optional) Is there anything else about this sample that you would like to add?') }}" rows=3 {% if sample.sample_edit_locked and not admin_mode %}disabled{% endif %}>{{ sample.sample_notes if sample.sample_notes is not none else '' |e}}</textarea>
+                            <textarea class="form-control" id="sample_notes" name="sample_notes" placeholder="{{ _('(Optional) Is there anything else about this sample that you would like to add?') }}" rows=3 {% if (sample.sample_edit_locked and not admin_mode) or ((need_reconsent_data or need_reconsent_biospecimen) and not admin_mode) %}disabled{% endif %}>{{ sample.sample_notes if sample.sample_notes is not none else '' |e}}</textarea>
```
Comment (wasade) [inline microsetta_interface/templates/sample.jinja2:244]: same here

**Review by wasade — CHANGES_REQUESTED** [review]

File: microsetta_interface/implementation.py [hunk 4]
```diff
@@ -1079,6 +1079,19 @@ def get_account(*, account_id=None):
...
+    # Determine if the source/profile has any action items. This framework
+    # will evolve over time.
+    for s in sources:
+        alerts = 0
+
+        need_reconsent_d = check_current_consent(
+            account_id, s[SOURCE_ID], "data"
+        )
+        if need_reconsent_d:
+            alerts += 1
```
[the remark this answers is not in the document]
Comment (cassidysymons) [reply microsetta_interface/implementation.py:1091]: Integer, and the UI does use it appropriately - https://github.com/biocore/microsetta-interface/pull/285/files#diff-3a82375541f7c64487a4c612c45b31cdecd3a80bdaf626e38b23e814420cee6fR60-R64

At the moment, it's impossible to be greater than 1, but it's establishing a framework that will be leveraged for future important updates.

File: microsetta_interface/implementation.py [hunk 5]
```diff
@@ -1362,6 +1385,11 @@ def get_fill_source_survey(*,
             ctr = 0
             trig_ctr = 0
             for field in group['fields']:
+                if need_reconsent:
+                    # If the user has not agreed to the current consent, we
+                    # disable all of the fields.
+                    field['disabled'] = True
```
[the remark this answers is not in the document]
Comment (cassidysymons) [reply microsetta_interface/implementation.py:1391]: Yes, disabled is included in all field types [coming out of the API](https://github.com/biocore/microsetta-private-api/blob/master/microsetta_private_api/model/vue/vue_field.py#L29).

[the remark this answers is not in the document]
Comment (wasade) [reply microsetta_interface/implementation.py:1391]: thanks!

File: microsetta_interface/implementation.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (wasade) [reply microsetta_interface/implementation.py:1091]: thanks!

File: microsetta_interface/templates/new_participant.jinja2 [hunk 6]
```diff
@@ -211,19 +211,27 @@
 {% block content%}
 
 <div class="container default-container">
-    {% if reconsent is not none %}
+    {% if reconsent != None %}
```
[the remark this answers is not in the document]
Comment (cassidysymons) [reply microsetta_interface/templates/new_participant.jinja2]: Adjusted

File: microsetta_interface/templates/new_results_page.jinja2 (hunk 1 above)
Comment (cassidysymons) [reply microsetta_interface/templates/new_results_page.jinja2]: Changed the verbiage

File: microsetta_interface/templates/sample.jinja2 (hunk 2 above)
Comment (cassidysymons) [reply microsetta_interface/templates/sample.jinja2]: Fixed

File: microsetta_interface/templates/sample.jinja2 (hunk 3 above)
Comment (cassidysymons) [reply microsetta_interface/templates/sample.jinja2]: Fixed

**Review by wasade — APPROVED** [review]

**Pull request merged**